cpp/src/arrow/compute/exec/asof_join_node.cc

```diff
@@ -380,6 +501,23 @@ class InputState {
     }
   }
 
+  static Result<std::unique_ptr<InputState>> Make(
+      size_t index, TolType tolerance, bool must_hash, bool may_rehash,
+      KeyHasher* key_hasher, ExecNode* node, ExecNode* output,
+      std::atomic<int32_t>& backpressure_counter,
+      const std::shared_ptr<arrow::Schema>& schema, const col_index_t time_col_index,
+      const std::vector<col_index_t>& key_col_index) {
+    constexpr size_t low_threshold = 4, high_threshold = 8;
```

Comment:
Do we want to keep these hard-coded in this PR?

Reply:
Ok for now. Let's try not make additional changes on top of the original one here.